src/components/map/layers/edit/EditLinkLayer.tsx

```diff
@@ -53,25 +53,37 @@ class EditLinkLayer extends Component<IEditLinkLayerProps> {
 
         map.off('editable:vertex:dragend');
         map.on('editable:vertex:dragend', () => {
-            const linkCopy = Object.assign({}, this.props.linkStore!.link);
-            const latlngs = this.editableLinks[0].getLatLngs()[0] as L.LatLng[];
-            linkCopy.geometry = latlngs;
-            this.props.linkStore!.setLink(linkCopy);
+            this.refreshEditableLink();
+        });
+
+        map!.off('editable:vertex:deleted');
+        map!.on('editable:vertex:deleted', (data: any) => {
+            this.refreshEditableLink();
         });
     }
 
+    private refreshEditableLink() {
+        const latlngs = this.editableLinks[0].getLatLngs()[0] as L.LatLng[];
+        this.props.linkStore!.changeLinkGeometry(latlngs);
+    }
+
     private drawEditableLinkToMap = (link: ILink) => {
         const map = this.props.leaflet.map;
         if (map) {
-            const editableLink = L.polyline([link.geometry]).addTo(map);
+            const editableLink = L.polyline(
+                [_.cloneDeep(link.geometry)],
+                { interactive: false },
+            ).addTo(map);
             editableLink.enableEdit();
             const latLngs = editableLink.getLatLngs() as L.LatLng[][];
             const coords = latLngs[0];
             const coordsToDisable = [coords[0], coords[coords.length - 1]];
             coordsToDisable.forEach((coordToDisable: any) => {
                 const vertexMarker = coordToDisable.__vertex;
                 vertexMarker.dragging.disable();
+                vertexMarker._events.click = {};
                 vertexMarker.setOpacity(0);
+                vertexMarker.setZIndexOffset(-1000);
```

Comment:
We should consider adding comments in sections as these, I have no idea what the code does just by looking at it.

Reply:
Added comment